### PR TITLE
WorldGuard compatibility

### PR DESCRIPTION
**Pull request by fastfend**
This PR adds compatibility with WorldGuard plugin.

When BetterPortals tries to create new portal using NewPortalChecker it checks for WG plugin and if it is enabled checks if player can build in desired location. If somehow player would not be provided default logic of WG is used

### CODE REVIEW

**Comment by Lauriethefish** [comment]
Might want to check [this post](https://worldguard.enginehub.org/en/latest/developer/dependency/) - as far as I know, with how you're doing things right now, if WG is not installed you'll get `ClassDefNotFoundError`s.

**Comment by Lauriethefish** [comment]
TL;DR, `canBuildInRegion` needs to be in an entirely separate class. You can then bind an implementation of that class that always returns true if WG is not installed, and otherwise you should bind the WG version of the class (using `Class.forName` in order to avoid directly referencing the class)

**Comment by fastfend** [comment]
Hi! In my implementation I use

`        if(Bukkit.getPluginManager().getPlugin("WorldGuard") == null) {
            return true;
        }
`

This way I never load class related to WG until next steps which ensures that WorldGuard is loaded. The problem you are talking about isn't going to happen because it will never go beyond that point. Java loads classes when necessary that's why it works. Tested and it works without WG (without errors in console). Even though if you want I can rewrite it to other class

**Comment by Lauriethefish** [comment]
Alright, if you're sure it works that's fine.
Could you think about implementing a custom WG flag? I think that'd be better than using block place.

**Comment by fastfend** [comment]
Sure! Definetly it will add more flexibilty for users. Also what do you think about resigning from forcing portal placement if no space was found?
Maybe condition it based on WG apperance too?

**Comment by Lauriethefish** [comment]
Hmm, I'll think about that. Performance is another thing that will need to be thought about - BP can check many thousands of portal positions on each spawn, especially with the new world height limit.

**Comment by fastfend** [comment]
Ok, I've added requested stuff. I've also made some performance tests and sadly when there is no space for portal check is running for about 10 sec on 5600X. I know that it's done off thread so world lives, but one core is sucked out of it all juice.
When more portals are ignited at the same time it still uses one core so it just makes search longer. Also there is about 1GB memory spike (mostly consisting of bukkit Location objects, but also BlockPosition).

tldr If portal is being ignited and there is big region which blocks it BetterPortals will use all out of one core

**Comment by fastfend** [comment]
Came up with idea of "optimization". BetterPortals will find candidate for portal position quickly at border of that WG region. To prevent unnecessary work from BP I'm gonna block portal ignition for scaled region (1/8x and 8x etc.). This way BP won't start search if portal is ignited in center of WG region
